airflow/serialization/helpers.py

```diff
@@ -37,7 +39,26 @@ def is_jsonable(x):
         else:
             return True
 
+    max_size = conf.getint("core", "max_templated_field_size")
+
     if not is_jsonable(template_field):
+        if isinstance(template_field, (list, tuple)):
```

Comment:
Why does this only check the first item?

Reply:
I would suggest to calculate the overall object size (summing the all the elements sizes) and compare it with the max size.

Reply:
Just did